### PR TITLE
Name Tags & Limit Dog Wander Radius

### PR DESCRIPTION
**Pull request by gracebarsny**
**Prefabs**
- Dog: new dog prefab with name tag
- HomeUI: uses copies of new dog prefab

**Scenes**
- DogWorld_Dev: testing for dog name tags

**Scripts**
- DogWorldSlot: added ref to name tag, added subscription methods, moved around setting of some refs
- NameTag: new script
- PPHomeUIController: subscribes to all the name tags so it can show the dog profile
- DogWorldTest: fleshed out test script to accommodate new functionality
- DogAI: cut the wander radius in half (should limit the dogs from climbing walls)

### CODE REVIEW

File: Assets/Scripts/UI/PickupPup/DogWorld/DogWorldSlot.cs [hunk 1]
```diff
@@ -8,13 +8,31 @@
...
+
     #region MonoBehaviourExtended Overrides 
 
     protected override void setReferences()
     {
         base.setReferences();
         dogImage = GetComponent<Image>();
+        nameTag = GetComponentInChildren<NameTag>();
+        Image[] images = GetComponentsInChildren<Image>();
+        if(images.Length == 1)
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogWorld/DogWorldSlot.cs]: May be cleaner to set this reference in the inspector

**Comment by imann24** [comment]
Possibly just the font, but the names appear very squished: 
image

**Comment by imann24** [comment]
Is it a bug that the name tags can render under other dogs?
image

**Comment by imann24** [comment]
Anchors for name tags are uniform instead of adjusting to the height of the dog sprite:

image

**Comment by imann24** [comment]
Very easy to select the wrong dog based on how Unity UI is defining the image bounds (clicked where my is in the screenshot): 
image

File: Assets/Scripts/UI/PickupPup/DogWorld/NameTag.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
+﻿/*
+ * Author(s): Grace Barrett-Snyder
+ * Description: Controls a dog's name tag above its dog world slot.
+ */
+using UnityEngine.UI;
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogWorld/NameTag.cs]: Believe we typically include another newline here

File: Assets/Scripts/World/PickupPup/DogAI/DogAI.cs [hunk 3]
```diff
@@ -119,7 +119,7 @@ IEnumerator wander()
             if (target == GetComponent<RectTransform>().anchoredPosition)
             {
                 float cTheta = UnityEngine.Random.Range(0, 2 * Mathf.PI);
-                float cRadius = Screen.width * Mathf.Sqrt(UnityEngine.Random.Range(0f, 1f));
+                float cRadius = Screen.width / 2 * Mathf.Sqrt(UnityEngine.Random.Range(0f, 1f));
```
Comment (imann24) [inline Assets/Scripts/World/PickupPup/DogAI/DogAI.cs]: Potentially worth making a tuning value, but could do that later

Comment (gracebarsny) [inline Assets/Scripts/World/PickupPup/DogAI/DogAI.cs]: How would you propose we do that? (As in, just the `/ 2` or the entire line?)

Comment (imann24) [inline Assets/Scripts/World/PickupPup/DogAI/DogAI.cs]: Yeah, just the `/ 2` as a scalar variable if we were to implement a tuning var there.

Comment (imann24) [inline Assets/Scripts/World/PickupPup/DogAI/DogAI.cs]: But I think ideally we'll need a more complex system than a radius to handle navigation, so likely not worth implementing

**Comment by gracebarsny** [comment]
@imann24 Cleaned up ref assignment in DogWorldSlot and made sure name tags always showed on top of surrounding dogs. The image bounds and spacing will have to wait until another sprint (dog sprite sizes are uniform right now regardless of size, making it trickier to adjust the name tag position according to size).

**Comment by imann24** [comment]
Nice! Looks good!

**Pull request merged**